gui/src/components/Dashboards/RecentActivity.js

```diff
@@ -0,0 +1,138 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import { withStyles } from '@material-ui/core/styles'
+import { fade } from '@material-ui/core/styles/colorManipulator'
+import Typography from '@material-ui/core/Typography'
+import Grid from '@material-ui/core/Grid'
+import Table from '@material-ui/core/Table'
+import TableBody from '@material-ui/core/TableBody'
+import TableCell from '@material-ui/core/TableCell'
+import TableRow from '@material-ui/core/TableRow'
+import CardContent from '@material-ui/core/CardContent'
+import Button from '@material-ui/core/Button'
+import Card from '@material-ui/core/Card'
+import ReactStaticLinkComponent from 'components/ReactStaticLinkComponent'
+import Link from 'components/Link'
+import TimeAgo from 'components/TimeAgo'
+import RunStatus from 'components/JobRuns/Status'
+
+const styles = theme => {
+  return {
+    cell: {
+      borderColor: theme.palette.divider,
+      borderTop: `1px solid`,
+      borderBottom: 'none',
+      padding: 0,
+      '&:hover': {
+        backgroundColor: fade(theme.palette.grey.A100, 0.2)
+      }
+    },
+    status: {
+      position: 'absolute',
+      top: 0,
+      left: 0,
+      paddingTop: 35,
+      paddingLeft: 35,
+      borderRight: 'solid 1px',
+      borderRightColor: theme.palette.divider,
+      width: 50,
+      height: '100%'
+    },
+    runDetails: {
+      padding: theme.spacing.unit * 4
+    }
+  }
+}
+
+const RecentActivity = ({classes, runs}) => {
+  let activity
+
+  if (!runs) {
```

Comment:
When would `runs` be nil or undefined? Can we coerce this into an array earlier in the call stack and be confident about it being an array moving forward? Or is this essentially a `loading` flag?

Reply:
It's the initial value and is used to display the loading indicator

Reply:
Can we place
```
let loading = !runs
```
above to improve readability?